src/view/trip-events-list-item-edit.js

```diff
@@ -160,6 +166,48 @@ export default class TripEventsEdit extends SmartView {
   restoreHandlers = () => {
     this.#setInnerHandlers();
     this.setListItemEditClickHandler(this._callback.editClick);
+    this.#setDatepicker();
+  }
+
+  removeElement = () => {
+    super.removeElement();
+
+    if (this.#datepicker) {
+      this.#datepicker.destroy();
+      this.#datepicker = null;
+    }
+  }
+
+  #setDatepicker = () => {
+    this.#datepicker = flatpickr(
+      this.element.querySelector('.event__input--time[name="event-start-time"]'),
+      {
+        dateFormat: 'd/m/Y H:i',
+        defaultDate: this.getData().startDate,
+        onChange: this.#dateStartChangeHandler,
+      },
+    );
```

Comment:
Борись за изолированные функции - в твоем проекте "заказчик" не интересуется с помощью чего ты сделал выбор даты. Зато ему интересно, чтобы форма выбора был одинаковый.

сделай функцию - в отдельном модуле

```javascript
const DATE_FORMAT = 'd/m/Y H:i'
const makePicker = (target, initialValue, onChange)=>flatpickr(target,{dateFormat:DATE_FORMAT, defaultDate: initialValue, onChange})
```

и тогда использовать в двух местах ее

```suggestion
    this.#datepicker = makePicker (
      this.element.querySelector('.event__input--time[name="event-start-time"]'),
      this.getData().startDate,
      this.#dateStartChangeHandler,
    );
```